server/src/com/thoughtworks/go/server/util/GoMacLauncher.java

```diff
@@ -41,76 +40,16 @@
     private Application application = Application.getApplication();
     private static final int MAX_SLEEP_SECONDS = 2400;
     private static final int SLEEP_MILLIS = 1000;
-    private static final String APPLICATION_SUPPORT_PATHNAME = System.getProperty("user.home")+"/Library/Application Support/Go Server/";
+    private static final String APPLICATION_SUPPORT_PATHNAME =
+            MessageFormat.format("{0}/Library/Application Support/{1}/", System.getProperty("user.home"), System.getProperty("go.application.name", "Go Server"));
     private static final String GO_CONFIG_DIRECTORY_PATH = APPLICATION_SUPPORT_PATHNAME + "config/";
     private static final String CRUISE_SERVER_URL = "http://localhost:8153/go/";
 
     public static void main(String[] args) {
         new File(GO_CONFIG_DIRECTORY_PATH).mkdirs();
-
-        //TODO: remove this call after a few releases
-        moveFilesFromGoServerDirToConfigDir();
```

Comment:
The files movement to config directory was done as a part of a bug fix(Old tracker #5985 - not able to take backup because backup never complete in macosx). Since you have removed moveFilesFromGoServerDirToConfigDir method, one cannot upgrade from Go (pre-12.2) to 15.x or later directly. They need to upgrade to a version between 12.2 and 14.4 first, and then to 15.1. This needs to be documented.